### ISSUE TITLE
powerdnsadmin.models.user.Anonymous object has no attribute e-mail

### ISSUE THREAD

**Issue by pcevela** [ngoduykhanh/PowerDNS-Admin#723]
## Error
I'm getting following error during loading of login page to syslog:
```
jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 134, in handle
    self.handle_request(listener, req, client, addr)
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.6/site-packages/werkzeug/middleware/proxy_fix.py", line 169, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1878, in handle_exception
    server_error = handler(server_error)
  File "/opt/powerdns-admin/powerdnsadmin/routes/base.py", line 28, in handle_internal_server_error
    return render_template('errors/500.html', code=500, message=e), 500
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/opt/powerdns-admin/powerdnsadmin/templates/errors/500.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/opt/powerdns-admin/powerdnsadmin/templates/base.html", line 34, in top-level template code
    {% set gravatar_url = current_user.email|email_to_gravatar_url(size=80) %}
  File "/opt/powerdns-admin/powerdnsadmin/lib/utils.py", line 187, in email_to_gravatar_url
    hash_string = hashlib.md5(email.encode('utf-8')).hexdigest()
jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
```
Other than this error, PDNS-A is running well
## Environment
* Fresh installation 9. 5. 2020
* CentOS 8
* Python 3.6
* yarn 1.22.4
* node 12.16.1
* Nginx > Gunicorn running as a service

**Comment by RoeiGanor** [comment]
I'm having this message as well and as I far as I can understand this bug is happening when the login page rendered before the user profile loads.
This error have never happened to me after the session created. I think you can generally ignore this in your deployment but if someone can find a fix that would be great.

**Comment by pcevela** [comment]
Yeah I agree, but it's creating a lot of noise in syslog, together with favicon.ico internal server error. I'll deal with favicon on nginx level, but this will need deeper look...

**Comment by johnwc** [comment]
Fresh install yesterday, and today I get a 'Please confirm your email address first'. Click 'Resend confirmation email' and get 'Internal Server Error' and the Anonymous object has no attribute email in the log.

**Comment by ngoduykhanh** [comment]
Interesting! The login page should not have anything to do with the user model. How did you get that error message? Maybe you can show me the step to reproduce so I can take a look and fix it.

**Comment by johnwc** [comment]
I did just what I wrote. I installed it one evening, logged in, created a domain to test. Came back the next morning, tried to login and a orange/red bar showed up above the fields that said I needed to confirm my email. I then clicked the Resend confirmation email link, and I recieved the blank page with the Internal Server Error. Looking in the logs, I see the Anonymous obj no email error.

**Comment by ngoduykhanh** [comment]
Thanks for the clarification. I can see the issue now. It was because the confirmation email template refer to the base template which has an unwanted user model call. I've submitted the fix in [this commit](https://github.com/ngoduykhanh/PowerDNS-Admin/commit/ab7e1eb71be39c8206f15120cd987883fb846280). The latest code on master branch should work properly now.

**Comment by pcevela** [comment]
Let me check on Monday also on my deployment, since I didn't had email verification (don't have smtp server configured) or didn't had red bar on top of login screen.

**Comment by sshipway** [comment]
Ive just upgraded to the current snapshot (21/5) from a previous one (21/4) and started to get this as well.  So it looks like something introduced in a recent commit.

**Comment by sshipway** [comment]
Looks like my problem was cured by a complete browser restart and cookies being cleared.  Something was invalidating the oAuth session but pdnsa didnt pick up on this, it just continues with no metadata and so fell over as there was no email or anything to work on.
I did initially try the suggested action to set the confirmed=1 on the user table but that didn't help, I suspect it only applies to internally-authed accounts.

**Comment by ngoduykhanh** [comment]
@sshipway the issue mentioned above isn't related to your problem I believe. Please open another issue thread if you find it.

**Comment by pcevela** [comment]
I can confirm [ab7e1eb](https://github.com/ngoduykhanh/PowerDNS-Admin/commit/ab7e1eb71be39c8206f15120cd987883fb846280) fixed reported issue issue. Now syslog is clean as whistle :)

**Comment by ngoduykhanh** [comment]
Thank you for your confirmation. Closing the issue.

**Comment by pcevela** [comment]
Sorry guys, I need to reopen it.
During testing API I figured out that not only login page generated this issue (which is now fixed), but also error templates/pages.
This was generated just by using/loginn in url, instead of /login

I was looking into fixing it same as you fixed login, but at this moment I'm more afraid to break something else since I don't know all relations.
```
May 22 12:09:33: [2020-05-22 12:09:33,673] [app.py:1891] ERROR - Exception on /loginn [GET]
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
May 22 12:09:33:     rv = self.dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1925, in dispatch_request
May 22 12:09:33:     self.raise_routing_exception(req)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1907, in raise_routing_exception
May 22 12:09:33:     raise request.routing_exception
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/ctx.py", line 350, in match_request
May 22 12:09:33:     result = self.url_adapter.match(return_rule=True)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/werkzeug/routing.py", line 1945, in match
May 22 12:09:33:     raise NotFound()
May 22 12:09:33: werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
May 22 12:09:33: During handling of the above exception, another exception occurred:
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
May 22 12:09:33:     response = self.full_dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
May 22 12:09:33:     rv = self.handle_user_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in handle_user_exception
May 22 12:09:33:     return self.handle_http_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1743, in handle_http_exception
May 22 12:09:33:     return handler(e)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/routes/base.py", line 24, in handle_page_not_found
May 22 12:09:33:     return render_template('errors/404.html', code=404, message=e), 404
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
May 22 12:09:33:     ctx.app,
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
May 22 12:09:33:     rv = template.render(context)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
May 22 12:09:33:     self.environment.handle_exception()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
May 22 12:09:33:     reraise(*rewrite_traceback_stack(source=source))
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
May 22 12:09:33:     raise value.with_traceback(tb)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/errors/404.html", line 1, in top-level template code
May 22 12:09:33:     {% extends "base.html" %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/base.html", line 34, in top-level template code
May 22 12:09:33:     {% set gravatar_url = current_user.email|email_to_gravatar_url(size=80) %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/lib/utils.py", line 187, in email_to_gravatar_url
May 22 12:09:33:     hash_string = hashlib.md5(email.encode('utf-8')).hexdigest()
May 22 12:09:33: jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
May 22 12:09:33: [2020-05-22 12:09:33 +0200] [17351] [ERROR] Error handling request /loginn
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
May 22 12:09:33:     rv = self.dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1925, in dispatch_request
May 22 12:09:33:     self.raise_routing_exception(req)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1907, in raise_routing_exception
May 22 12:09:33:     raise request.routing_exception
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/ctx.py", line 350, in match_request
May 22 12:09:33:     result = self.url_adapter.match(return_rule=True)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/werkzeug/routing.py", line 1945, in match
May 22 12:09:33:     raise NotFound()
May 22 12:09:33: werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
May 22 12:09:33: During handling of the above exception, another exception occurred:
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
May 22 12:09:33:     response = self.full_dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
May 22 12:09:33:     rv = self.handle_user_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in handle_user_exception
May 22 12:09:33:     return self.handle_http_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1743, in handle_http_exception
May 22 12:09:33:     return handler(e)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/routes/base.py", line 24, in handle_page_not_found
May 22 12:09:33:     return render_template('errors/404.html', code=404, message=e), 404
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
May 22 12:09:33:     ctx.app,
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
May 22 12:09:33:     rv = template.render(context)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
May 22 12:09:33:     self.environment.handle_exception()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
May 22 12:09:33:     reraise(*rewrite_traceback_stack(source=source))
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
May 22 12:09:33:     raise value.with_traceback(tb)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/errors/404.html", line 1, in top-level template code
May 22 12:09:33:     {% extends "base.html" %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/base.html", line 34, in top-level template code
May 22 12:09:33:     {% set gravatar_url = current_user.email|email_to_gravatar_url(size=80) %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/lib/utils.py", line 187, in email_to_gravatar_url
May 22 12:09:33:     hash_string = hashlib.md5(email.encode('utf-8')).hexdigest()
May 22 12:09:33: jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
May 22 12:09:33: During handling of the above exception, another exception occurred:
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 134, in handle
May 22 12:09:33:     self.handle_request(listener, req, client, addr)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
May 22 12:09:33:     respiter = self.wsgi(environ, resp.start_response)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2463, in __call__
May 22 12:09:33:     return self.wsgi_app(environ, start_response)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/werkzeug/middleware/proxy_fix.py", line 169, in __call__
May 22 12:09:33:     return self.app(environ, start_response)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2449, in wsgi_app
May 22 12:09:33:     response = self.handle_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1878, in handle_exception
May 22 12:09:33:     server_error = handler(server_error)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/routes/base.py", line 28, in handle_internal_server_error
May 22 12:09:33:     return render_template('errors/500.html', code=500, message=e), 500
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
May 22 12:09:33:     ctx.app,
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
May 22 12:09:33:     rv = template.render(context)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
May 22 12:09:33:     self.environment.handle_exception()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
May 22 12:09:33:     reraise(*rewrite_traceback_stack(source=source))
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
May 22 12:09:33:     raise value.with_traceback(tb)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/errors/500.html", line 1, in top-level template code
May 22 12:09:33:     {% extends "base.html" %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/base.html", line 34, in top-level template code
May 22 12:09:33:     {% set gravatar_url = current_user.email|email_to_gravatar_url(size=80) %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/lib/utils.py", line 187, in email_to_gravatar_url
May 22 12:09:33:     hash_string = hashlib.md5(email.encode('utf-8')).hexdigest()
May 22 12:09:33: jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
May 22 12:09:33: [2020-05-22 12:09:33,933] [app.py:1891] ERROR - Exception on /favicon.ico [GET]
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
May 22 12:09:33:     rv = self.dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1925, in dispatch_request
May 22 12:09:33:     self.raise_routing_exception(req)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1907, in raise_routing_exception
May 22 12:09:33:     raise request.routing_exception
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/ctx.py", line 350, in match_request
May 22 12:09:33:     result = self.url_adapter.match(return_rule=True)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/werkzeug/routing.py", line 1945, in match
May 22 12:09:33:     raise NotFound()
May 22 12:09:33: werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
May 22 12:09:33: During handling of the above exception, another exception occurred:
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
May 22 12:09:33:     response = self.full_dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
May 22 12:09:33:     rv = self.handle_user_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in handle_user_exception
May 22 12:09:33:     return self.handle_http_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1743, in handle_http_exception
May 22 12:09:33:     return handler(e)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/routes/base.py", line 24, in handle_page_not_found
May 22 12:09:33:     return render_template('errors/404.html', code=404, message=e), 404
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
May 22 12:09:33:     ctx.app,
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
May 22 12:09:33:     rv = template.render(context)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
May 22 12:09:33:     self.environment.handle_exception()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
May 22 12:09:33:     reraise(*rewrite_traceback_stack(source=source))
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
May 22 12:09:33:     raise value.with_traceback(tb)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/errors/404.html", line 1, in top-level template code
May 22 12:09:33:     {% extends "base.html" %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/base.html", line 34, in top-level template code
May 22 12:09:33:     {% set gravatar_url = current_user.email|email_to_gravatar_url(size=80) %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/lib/utils.py", line 187, in email_to_gravatar_url
May 22 12:09:33:     hash_string = hashlib.md5(email.encode('utf-8')).hexdigest()
May 22 12:09:33: jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
May 22 12:09:33: [2020-05-22 12:09:33 +0200] [17351] [ERROR] Error handling request /favicon.ico
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
May 22 12:09:33:     rv = self.dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1925, in dispatch_request
May 22 12:09:33:     self.raise_routing_exception(req)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1907, in raise_routing_exception
May 22 12:09:33:     raise request.routing_exception
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/ctx.py", line 350, in match_request
May 22 12:09:33:     result = self.url_adapter.match(return_rule=True)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/werkzeug/routing.py", line 1945, in match
May 22 12:09:33:     raise NotFound()
May 22 12:09:33: werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
May 22 12:09:33: During handling of the above exception, another exception occurred:
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
May 22 12:09:33:     response = self.full_dispatch_request()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
May 22 12:09:33:     rv = self.handle_user_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in handle_user_exception
May 22 12:09:33:     return self.handle_http_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1743, in handle_http_exception
May 22 12:09:33:     return handler(e)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/routes/base.py", line 24, in handle_page_not_found
May 22 12:09:33:     return render_template('errors/404.html', code=404, message=e), 404
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
May 22 12:09:33:     ctx.app,
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
May 22 12:09:33:     rv = template.render(context)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
May 22 12:09:33:     self.environment.handle_exception()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
May 22 12:09:33:     reraise(*rewrite_traceback_stack(source=source))
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
May 22 12:09:33:     raise value.with_traceback(tb)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/errors/404.html", line 1, in top-level template code
May 22 12:09:33:     {% extends "base.html" %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/base.html", line 34, in top-level template code
May 22 12:09:33:     {% set gravatar_url = current_user.email|email_to_gravatar_url(size=80) %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/lib/utils.py", line 187, in email_to_gravatar_url
May 22 12:09:33:     hash_string = hashlib.md5(email.encode('utf-8')).hexdigest()
May 22 12:09:33: jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
May 22 12:09:33: During handling of the above exception, another exception occurred:
May 22 12:09:33: Traceback (most recent call last):
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 134, in handle
May 22 12:09:33:     self.handle_request(listener, req, client, addr)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
May 22 12:09:33:     respiter = self.wsgi(environ, resp.start_response)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2463, in __call__
May 22 12:09:33:     return self.wsgi_app(environ, start_response)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/werkzeug/middleware/proxy_fix.py", line 169, in __call__
May 22 12:09:33:     return self.app(environ, start_response)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2449, in wsgi_app
May 22 12:09:33:     response = self.handle_exception(e)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1878, in handle_exception
May 22 12:09:33:     server_error = handler(server_error)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/routes/base.py", line 28, in handle_internal_server_error
May 22 12:09:33:     return render_template('errors/500.html', code=500, message=e), 500
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
May 22 12:09:33:     ctx.app,
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
May 22 12:09:33:     rv = template.render(context)
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
May 22 12:09:33:     self.environment.handle_exception()
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
May 22 12:09:33:     reraise(*rewrite_traceback_stack(source=source))
May 22 12:09:33:   File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
May 22 12:09:33:     raise value.with_traceback(tb)
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/errors/500.html", line 1, in top-level template code
May 22 12:09:33:     {% extends "base.html" %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/templates/base.html", line 34, in top-level template code
May 22 12:09:33:     {% set gravatar_url = current_user.email|email_to_gravatar_url(size=80) %}
May 22 12:09:33:   File "/opt/powerdns-admin/powerdnsadmin/lib/utils.py", line 187, in email_to_gravatar_url
May 22 12:09:33:     hash_string = hashlib.md5(email.encode('utf-8')).hexdigest()
May 22 12:09:33: jinja2.exceptions.UndefinedError: 'powerdnsadmin.models.user.Anonymous object' has no attribute 'email'
```

**Comment by ngoduykhanh** [comment]
It seems like the original issue comes from `current_user.email` usage in the base template.  I've submitted a quick fix for it a598c527299e172e68d0988b00472e3595673e72. Let me know if it solves the issue. Thanks for reporting @pcevela

**Comment by pcevela** [comment]
I think we are good - I don't see errors in syslog or Internal Server Error anymore.
Thanks for quick fix! :)
